src/boat/private/BoatConfig.nim

```diff
@@ -0,0 +1,100 @@
+##
+## Config
+## ======
+## extendable interface for creating, parsing and saving boat configs
+
+##[
+## TLDR
+- come back later
+
+todos
+-----
+- cmd to parse arbitrarily, e.g. boat conf load ./some/dir
+- cmd to see current captains: e.g. boat conf list -> X, Y, Z
```

Comment:
theres a fn in FileManager for parsing configs
and you should be able to query captainslog for any information